### ISSUE TITLE
Incorrectly finds "Type[Enum]" is not iterable

### ISSUE THREAD

**Issue by ironfoot** [microsoft/pyright#1050]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
pyright incorrectly thinks enums are not iterable.
It only happens when the variable is a member variable.
Doesn't happen when the variable is a function argument.

**To Reproduce**

```python
class Color(enum.Enum):
    RED = enum.auto()
    GREEN = enum.auto()


class Foo:
    _foo: Type[enum.Enum]

    def __init__(self):
        self._foo = Color

    def _print_foo(self):
        for f in self._foo:  # "Type[Enum]" is not iterable
            print(f)


def _test_fn(foo: Type[enum.Enum]) -> None:
    for f in foo:  # This one is fine.
        print(f)
```

**Expected behavior**
No error in both case.

**VS Code extension or command-line**
Through Pylance VS Code extension
Version: 2020.9.5

**Comment by erictraut** [comment]
Thanks for reporting the problem. This will be fixed in the next release of Pyright and Pylance.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.74, which I just published. It will also be included in the next release of Pylance.
